.github/workflows/misc-migtests.yml

```diff
@@ -69,6 +106,10 @@ jobs:
 
       - name: Run installer script to setup voyager
         run: |
+          # Clear Go build cache to ensure fresh compilation from current source
```

Comment:
I did not understand this. Why is there a need for clearing the cache in the first place? 

Which previous step would have caused caching?